src/Ocelot/DependencyInjection/OcelotBuilder.cs

```diff
@@ -74,6 +72,7 @@ public OcelotBuilder(IServiceCollection services, IConfiguration configurationRo
             Services.TryAddSingleton<IAuthenticationOptionsCreator, AuthenticationOptionsCreator>();
             Services.TryAddSingleton<IUpstreamTemplatePatternCreator, UpstreamTemplatePatternCreator>();
             Services.TryAddSingleton<IRequestIdKeyCreator, RequestIdKeyCreator>();
+            Services.TryAddSingleton<ITimeoutCreator, TimeoutCreator>();
```

Comment:
Who knows, maybe we will introduce new feature: Default timeout overriding 😉

Reply:
> Who knows, maybe we will introduce new feature: Default timeout overriding 😉

I added)

Reply:
@RaynaldM @ggnaegi This is major issue in my opinion!
We need to have consensus here. Do we add it to DI or not to add?
Will we allow to developers to override default timeout logic or not to allow?

Reply:
I agree, but we have to be careful not to change expected behavior (Open-closed principle), unless it's really necessary.
The idea is that, if I update the package, it continues to work without any change on my side (unless there's a breaking change).

Reply:
@hogwartsdeveloper @RaynaldM @ggnaegi 🆗 guys
Current votes are
- @raman-m ➖ disagree
- @ggnaegi ➕ agree, proposed `ITimeoutCreator` interface
- @RaynaldM ➕ agree

Reply:
@hogwartsdeveloper
Please delete this line as we are phasing out `ITimeoutCreator` in favor of `IRoutesCreator`. 
The testing of logic will be conducted within the `RoutesCreator` class.

P.S. `IRoutesCreator` has already been incorporated into DI.